File: app/src/main/java/com/bigscreen/mangindo/network/module/MangaModule.java
package com.bigscreen.mangindo.network.module;

import com.bigscreen.mangindo.network.service.MangaApiService;
import com.bigscreen.mangindo.network.service.MangaNetworkService;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import retrofit2.Retrofit;

@Module
public class MangaModule {

    @Provides
    @Singleton
    @SuppressWarnings("unused")
    public MangaNetworkService providesMangaNetworkService(Retrofit retrofit) {
        return retrofit.create(MangaNetworkService.class);
    }

    @Provides
    @Singleton
    @SuppressWarnings("unused")
    public MangaApiService providesMangaApiService(MangaNetworkService mangaNetworkService) {
        return new MangaApiService(mangaNetworkService);
    }
}

File: app/src/main/java/com/bigscreen/mangindo/deps/MangaModule.kt
package com.bigscreen.mangindo.deps

import com.bigscreen.mangindo.network.service.MangaApiService
import com.bigscreen.mangindo.network.service.MangaNetworkService
import dagger.Module
import dagger.Provides
import retrofit2.Retrofit
import javax.inject.Singleton

@Module
class MangaModule {

    @Provides
    @Singleton
    fun providesMangaNetworkService(retrofit: Retrofit): MangaNetworkService {
        return retrofit.create(MangaNetworkService::class.java)
    }

    @Provides
    @Singleton
    fun providesMangaApiService(mangaNetworkService: MangaNetworkService): MangaApiService {
        return MangaApiService(mangaNetworkService)
    }
}
